### DISCUSSION TITLE
Is RsBuild production ready and tree shakeable?

### DISCUSSION THREAD

**Discussion by AsuraKev** [web-infra-dev/modern.js#5070] [Q&A] [answered]
Stumble across this tool and wondering if its production ready and support treeshaking out of the box? Does it use RsPack as the bundler?

thanks

**Comment by chenjiahan** [comment, marked as answer]
Hi, Rsbuild is production ready now, a few APIs might change before v1.0.0. We plan to release v1.0 in Q1/Q2 2024.

It supports tree shaking and uses Rspack as the bundler by default (that's why it is called Rsbuild, haha)

**Reply by AsuraKev** [reply]
thanks!

**Reply by AsuraKev** [reply]
sorry just realised I was meant to ask about Modernjs/module, because I wanted to create a react component library with it. is it production ready and tree shakeable? :)

**Reply by chenjiahan** [reply]
yes, Modern.js Module is stable and a good choice to build React component lib 😄

**Reply by AsuraKev** [reply]
thanks! I am wondering if it can handle CSS in JS as well as Sub path entry? like this in package json

"exports": {
  "./core/*": "./dist/core/*.js",
  "./styles/*": "./dist/styles/*.js",
  "./package.json": "./package.json"
},

**Reply by chenjiahan** [reply]
Which CSS-in-JS library are you using? If the library does not require extra build configs or steps, it can be used in Modern.js Module out-of-the-box.
